src/utils/api/api.ts

```diff
@@ -102,6 +102,15 @@ function buildBody(options: IAPICall) {
     return body;
   }
 
+  if (options.requestType === "getBlocks") {
```

Comment:
these if statements work here technically but they aren't a great practice to have here imo. i believe there is an issue already for this however, so it should be cleaned up when that issue is resolved.

another process indicator of this for me: api is a fairly generic way of making api calls, but it has specific api call logic in it, which makes it not generic anymore. it also requires you to update api every time a new call is added, which can lead to breaking other api calls if the logic is somehow wrong (pretty rare but it can happen)

lastly, there is probably an eslint warning about redeclaring the payload variable. it might not error because it's used in proper scope in both places but usually warnings exist about this kind of thing.